extensions/crypto/lib.rs

```diff
@@ -234,210 +165,79 @@ pub async fn op_webcrypto_generate_key(
-      let private_key: EcdsaKeyPair = tokio::task::spawn_blocking(
-        move || -> Result<EcdsaKeyPair, ring::error::Unspecified> {
+      let private_key: Vec<u8> = tokio::task::spawn_blocking(
+        move || -> Result<Vec<u8>, ring::error::Unspecified> {
           let curve = curve.unwrap();
           let pkcs8 = EcdsaKeyPair::generate_pkcs8(curve, &rng)?;
-          Ok(EcdsaKeyPair::from_pkcs8(&curve, pkcs8.as_ref())?)
+          Ok(pkcs8.as_ref().to_vec())
         },
       )
       .await
       .unwrap()?;
 
-      // Create webcrypto keypair.
-      let webcrypto_key_public = WebCryptoKey::new_public(
-        algorithm,
-        extractable,
-        args.key_usages.clone(),
-      );
-      let webcrypto_key_private =
-        WebCryptoKey::new_private(algorithm, extractable, args.key_usages);
-      let crypto_key = WebCryptoKeyPair::new(
-        webcrypto_key_public,
-        webcrypto_key_private.clone(),
-      );
-
-      let rid = state.resource_table.add(CryptoKeyResource {
-        crypto_key: webcrypto_key_private,
-        key: private_key,
-        hash: args.algorithm.hash,
-      });
-
-      JsCryptoKey::Pair {
-        key: crypto_key,
-        private_rid: rid,
-        // NOTE: We're using the same Resource for public and private key since they are part
-        //       of the same interface in `ring`.
-        public_rid: rid,
-      }
+      private_key
     }
     Algorithm::Hmac => {
-      validate_usage!(&args.key_usages, vec![KeyUsage::Sign, KeyUsage::Verify]);
+      let hash: HmacAlgorithm = args.hash.ok_or_else(not_supported)?.into();
 
-      let hash: HmacAlgorithm =
-        args.algorithm.hash.ok_or_else(not_supported)?.into();
       let rng = RingRand::SystemRandom::new();
-      let key: HmacKey = tokio::task::spawn_blocking(
-        move || -> Result<HmacKey, ring::error::Unspecified> {
-          HmacKey::generate(hash, &rng)
-        },
-      )
-      .await
-      .unwrap()?;
+      let mut key_bytes = [0; ring::digest::MAX_OUTPUT_LEN];
+      let key_bytes = &mut key_bytes[..hash.digest_algorithm().output_len];
+      rng.fill(key_bytes)?;
 
-      let crypto_key =
-        WebCryptoKey::new_secret(algorithm, extractable, args.key_usages);
-      let resource = CryptoKeyResource {
-        crypto_key: crypto_key.clone(),
-        key,
-        hash: args.algorithm.hash,
-      };
-      JsCryptoKey::Single {
-        key: crypto_key,
-        rid: state.resource_table.add(resource),
-      }
+      key_bytes.to_vec()
     }
     _ => return Err(not_supported()),
   };
 
-  Ok(GenerateKeyResult { key })
+  Ok(key.into())
+}
+#[derive(Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum KeyFormat {
+  Raw,
+  Pcks8,
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub struct KeyData {
+  r#type: KeyFormat,
+  data: ZeroCopyBuf,
 }
 
 #[derive(Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct WebCryptoSignArg {
```

Comment:
Consistency nit: can we remove the Web prefix on these or is there a clash?

Reply:
Done :+1: 
1. removed Web prefix and instead used Crypto for generic names (eg: `CryptoHash` instead of `WebCryptoHash` or `Hash`)
2. argument structs are ending with `Arg` to avoid conflicts (eg: `AlgorithmArg` and not just `Algorithm` (conflicting))
3. renamed ops to follow `op_crypto_` pattern